torch/csrc/jit/passes/graph_fuser.cpp

```diff
@@ -83,12 +83,12 @@ bool isSimpleMap(Node *node) {
     return false;
   // Make sure that the node doesn't broadcast.
   JIT_ASSERT(node->inputs().size() > 0);
-  TensorType* expected_type = node->inputs()[0]->type()->cast<TensorType>();
+  TensorTypePtr expected_type = node->inputs()[0]->type()->cast<TensorType>();
   if (!expected_type) return false;
 //type checking is intentionally dropped from isSimpleMap
 //isFusable is checking input/output types as there are some exceptions from allFloatIO requirement
-  static const auto equal_modulo_strides = [](TensorType* expected, const TypePtr& _actual) {
-     TensorType* actual = _actual->cast<TensorType>();
+  static const auto equal_modulo_strides = [](TensorTypePtr expected, const TypePtr& _actual) {
```

Comment:
`expected` can be a `const&` too. Otherwise you'll unnecessarily bump the refcount at every loop iteration.